modules/openapi-generator/src/main/resources/kotlin-client/infrastructure/RequestConfig.kt.mustache

```diff
@@ -12,6 +12,7 @@ package {{packageName}}.infrastructure
     val method: RequestMethod,
     val path: String,
     val headers: MutableMap<String, String> = mutableMapOf(),
+    val params: MutableMap<String, Any> = mutableMapOf(),
```

Comment:
@stefankoppier Hi, thanks for creating this PR.
Why did you add this here? 
This alone doesn't do anything, or does it?

Reply:
All generators use the same infrastructure. 

I moved the substitution of path variables to the later stage in ApiClient (https://github.com/OpenAPITools/openapi-generator/pull/17493/files#diff-54ee9559f6e8a02839cafd21addca42b84d832f55df9870e33002e8f5148d963R44) to let Spring handle the encoding.

This does mean that the field params is not used in the other libraries.

Reply:
@4brunu @wing328 it would be nice if this PR could be included in 7.3.0 as it is a fairly annoying bug. If you want any changes, please let me know.

Reply:
@stefankoppier we definitely want to include your fix.

can you please PM me via Slack when you've time?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

Reply:
as discussed, it would be great if you can file another PR to add a test or 2 using the echo api spec.